crates/oxc_linter/src/lint_adapter.rs

```diff
@@ -0,0 +1,1208 @@
+use std::{cell::RefCell, collections::BTreeMap, rc::Rc, sync::Arc};
+
+use oxc_ast::{
+    ast::{
+        Argument, ArrayPattern, AssignmentExpression, AssignmentPattern, AssignmentTarget,
+        AssignmentTargetPattern, BindingPatternKind, Expression, FormalParameters,
+        ImportDeclarationSpecifier, ImportOrExportKind, JSXClosingElement, JSXElementName,
+        JSXOpeningElement, ModuleDeclaration, ObjectPattern, ObjectPropertyKind, PropertyKey,
+        SimpleAssignmentTarget, StringLiteral, TSLiteral, TSSignature, TSType, TSTypeAnnotation,
+        TSTypeName, TemplateLiteral, VariableDeclarator,
+    },
+    AstKind,
+};
+use oxc_semantic::{AstNode, Semantic};
+use oxc_span::{Atom, GetSpan, Span};
+use oxc_syntax::operator::BinaryOperator;
+use serde::Deserialize;
+use trustfall::{
+    provider::{
+        resolve_coercion_with, resolve_neighbors_with, resolve_property_with, Adapter,
+        ContextIterator, ContextOutcomeIterator, EdgeParameters, ResolveEdgeInfo, ResolveInfo,
+        TrustfallEnumVertex, VertexIterator,
+    },
+    FieldValue,
+};
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct InputQuery {
+    pub query: String,
+    pub args: BTreeMap<Arc<str>, FieldValue>,
+    pub reason: String,
+}
+
+#[derive(Debug, Clone, TrustfallEnumVertex)]
+pub enum Vertex<'a> {
+    AstNode(AstNode<'a>),
+    Expression(&'a Expression<'a>),
+    Argument(&'a Argument<'a>, i32), // i32 is argument index
+    Span(Span),
+    Parameter(&'a BindingPatternKind<'a>, i32), // i32 is argument index
```

Comment:
Consider making structs with named fields instead of relying on comments to communicate what the various positions mean. It pays off big time in the longer run.

Reply:
Definitely will do it along with stubgen transformation